src/net/RTP/RTPSession.cs

```diff
@@ -1033,6 +1087,44 @@ private SDP GetSessionDesciption(List<MediaStreamTrack> tracks, IPAddress connec
                     }
                 }
 
+                if (IsSecure)
+                {
+                    if (sdpType == SdpType.offer )
+                    {
+                        uint tag = 1;
+                        foreach (SDPSecurityDescription.CryptoSuites cryptoSuite in SrtpCryptoSuites)
+                        {
+                            announcement.SecurityDescriptions.Add(SDPSecurityDescription.CreateNew(tag, cryptoSuite));
+                            tag++;
+                        }
+                    }
+                    else
+                    {
+                        var sel = RemoteDescription?.Media.FirstOrDefault(a => a.MLineIndex == mindex)?.SecurityDescriptions
+                                                          .FirstOrDefault(s => SrtpCryptoSuites.Contains(s.CryptoSuite));
+                        
+                        if (sel == null)
+                        {
+                            throw new ApplicationException("Error creating crypto attribute. No compatible offer.");
+                        }
+                        else
+                        {
+                            announcement.SecurityDescriptions.Add(SDPSecurityDescription.CreateNew(sel.Tag, sel.CryptoSuite));
+                        }
+                    }
+
+                    m_srtpHandler.SetupLocal(announcement.SecurityDescriptions, sdpType);
+
+                    if (m_srtpHandler.IsNegotiationComplete)
+                    {
+                        SetSecurityContext(
+                            m_srtpHandler.ProtectRTP,
+                            m_srtpHandler.UnprotectRTP,
+                            m_srtpHandler.ProtectRTCP,
+                            m_srtpHandler.UnprotectRTCP);
+                    }
+                }
+
                 sdp.Media.Add(announcement);

```

Comment:
As above, `IsSecure` can also indicate a WebRTC connection which won't have these SDP attributes.

Reply:
Sorry, I missed the WebRTC things. I was focused to much on the SIP part.
I would suggest using a separate  variable in RtpSession to differentiate between SRTP for WebRTC and SIP.
I have changed IsSecure to UseSdpCryptoNegotiation for this purpose in my environment and can now succuessfully run the WebRTC integration tests.
What do you think?

Reply:
Yes, that sounds like a reasonable solution.